Artroom1/Controller/ArtViewController.swift

```diff
@@ -22,25 +22,46 @@ class ArtViewController: UIViewController {
     private var selection: [Artwork] = []
     private var animator: UIDynamicAnimator!
     private var snapping: UISnapBehavior!
+    
+    var selectedIndex: Int!
    
     func setCurrentArtwork(_ artwork: Artwork) {
         self.artResults.text = artwork.title
-        self.artImage.image = UIImage (named: artwork.image)
+        // instead of assigning an outlet like above, here I'm calling a function to do that - which assigns artImage and returns nothing
+        loadImageFromURL(artwork.image.url)
```

Comment:
Later I'll show you a small problem with this part of the code. But maybe you already want to think about it. Hint: Imagine the internet connection is reeeeeally slow and the user is swiping fast, what will happen?

Reply:
I'm trying something :) I think I know!

Reply:
Idea: Can I create a dictionary of images during the first loading process, store them, and just call on them later when I'm assigning?

Reply:
Tried something. Maybe you can show me if I went down the right track or completely off. :)

Reply:
As I wrote in the comment above, it went into a wrong direction :) What I want to say is that if you have a slow connection, you start a network call, because the user wants to see one image. While it's loading, the user swipes to the next image. So there will be a second network request starting. If now the second one (so from the image the user is currently supposed to see) finishes first, the user will see the correct image for some seconds. Now imagine the request from the first image finishes. With the current code, the image will be replaced and the user will see the wrong image.
I'll give you another hint: `URLSessionTask` has a method `cancel`.

Reply:
New attempt - it seems to work so far? If this is the smart thing to do I'll work it into ArtworksDatabase too.